packages/jest-cli/src/TestSequencer.js

```diff
@@ -59,6 +62,30 @@ export default class TestSequencer {
     return cache;
   }
 
+  _fileSizeRecurseDependencies(
+    test: Test,
+    stats: {[path: Path]: number},
+  ): number {
+    const {
+      path,
+      context: {resolver, hasteFS, config},
+    } = test;
+
+    const fileSize = path =>
+      stats[path] || (stats[path] = fs.statSync(path).size);
```

Comment:
I know we `stat`-ed previously as well, but would it be possible to look in `hasteFs` instead? I know it doesn't contain that metadata, but we could populate it during crawl (node crawler gets `stat` and the watchman crawler contains `size` in its results). During `change`, we get the `stat` object object, but we just use/keep `mtime`.

Then we avoid hitting the FS here, although we'll use more memory

Reply:
That sounds reasonable to me. Should we perhaps open an issue and do this in a follow-up PR though? Not sure how complicated it would be, I'm worried that this PR might get too large to handle, plus landing the hasteFS thing is kind of independent of changing the heuristic.